[[ Bugfix 11277 ]] Fix for develop-7.0

livecodesebastien opened the pull request:
This branch merges bugfix-11277_6.7 into develop-7.0, to make the develop-6.7 -> develop-7.0 merge simpler.

livecodesebastien committed:
Merge remote-tracking branch 'mark/bugfix-11277' into bugfix-11277_6.7

Conflicts:
	engine/src/cmdse.cpp
	engine/src/funcs.cpp

livecodesebastien committed:
[[ Bug 11277 ]] Implement "do", "value" and "merge" functions for Handlerlist (called in out-of-handler scripts)

livecodesebastien committed:
[[ Bug 11277 ]] Fix compilation issue

livecodesebastien committed:
Merge remote-tracking branch 'sebastien/bugfix-11277_6.7' into bugfix-11277_7.0

runrevmark commented:
I think you may want to redo this PR slightly once the changes to the 6.7 version I suggested have been done (i.e. the doscript / eval calls will end up in MCExecContext).

Also - can you double-check that each area which is listed is fixed in both 6.7 and 7.0 - this PR seems lighter than the 6.7 one...

livecodesebastien commented:
It is lighter for a simple reason: your branch bugfix-11277 had been merged in develop-7.0 and refactored (whereas it never made its way in develop-6.7), so the main part of the fixes were already in 7.0.
I'll probably close this pull request once I've changed the 6.7 one, since it is a merge-then-refactor of bugfix-11277_6.7 in develop-7.0.

livecodesebastien commented:
Closed, the merge will be done from merging develop-6.7 into develop-7.0
